app/models/job.ts

```diff
@@ -428,6 +442,51 @@ export class Job extends Record {
     return data.length !== 0;
   }
 
+  /**
+   * Determine if the current job is the same as an already running job. If so, replace the
+   * current job's requestId with that of the running job's. Add an attachedStatus property for
+   * later referencing.
+   *
+   * @param transaction - The transaction to use for saving the job
+   * @throws {@link Error} if the job is invalid
+   */
+  async maybeAttach(transaction: Transaction): Promise<void> {
+    const rows = await transaction('jobs')
+      .select('requestId', 'shapeFileUrl')
+      .where({ username: this.username, request: this.request })
+      .andWhere('status', JobStatus.ACCEPTED)
+      .orWhere('status', JobStatus.RUNNING);
+    if (Array.isArray(rows) && rows.length > 0) {
+      const originalId = this.requestId;
+      // if a shapefile is not present then database query returned all matching running requests
+      if (this.shapeFileUrl === '') {
+        const assumedId = rows.shift()['requestId'];
+        this.attachedStatus = { didAttach: true, originalId, assumedId };
+      // otherwise, compare checksum of current shapefile to other matching requests shapefile
+      } else {
+        const store = defaultObjectStore();
+        const thisSum = await objectStoreChecksum(store, this.shapeFileUrl);
+        let otherSum = '';
+        let otherId = '';
+        for (const row of rows) {
```

Comment:
Nit: this might be a little more concise (and functional, if that matters) if you used `Array.find` here instead of a `for` loop.

Reply:
I went with a `for` loop instead of `Array.find` because I wanted the ability to `break` early / not process the whole sequence necessarily.

Reply:
Array.find stops after the first success

Reply:
This code has been changed too: db rows object is `shift()`ed to if one or more matches are found.